commons/zenoh-buffers/src/slice.rs

```diff
@@ -242,12 +201,10 @@ impl<'a> SiphonableReader for &'a [u8] {
     where
         W: Writer,
     {
-        let res = writer.write(self).map_err(|_| DidntSiphon);
-        if let Ok(len) = res {
-            // SAFETY: len is returned from the writer, therefore it means
-            //         len amount of bytes have been written to the slice.
-            *self = crate::unsafe_slice!(self, len.get()..);
-        }
-        res
+        let len = writer.write(self).map_err(|_| DidntSiphon)?;
+        // SAFETY: len is returned from the writer, therefore it means
```

Comment:
We cannot rely on the correctness of the `Writer`. This should be unsafe as well. Otherwise `Writer` has to be an unsafe trait.

Reply:
As long as the trait is public, yes. Otherwise, it's better, but not really mandatory to me.

Reply:
I will rollback the change, as it has to be part of a bigger change (making `Writer` unsafe)